### PR TITLE
- More details about the integration with Amazon ecs.

### PR DESCRIPTION
**Pull request by rjschwei**
- Document the use of the SUSE provided pre-built docker images

### CODE REVIEW

File: xml/art_sle_amazon_ecs.xml [hunk 1]
```diff
@@ -41,21 +41,27 @@
     <para>
       Creating a new ECS cluster in Amazon is accomplished by using the
       <link xlink:href="https://console.aws.amazon.com/ecs/home#/firstRun">First
-      Run Wizard</link>
+      Run Wizard.</link> Alternatively you may use the
+      <emphasis role="strong">aws</emphasis> command line tools to
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Use `<command>` instead of `<emphasis>`.

File: xml/art_sle_amazon_ecs.xml [hunk 2]
```diff
@@ -41,21 +41,27 @@
...
+      <link xlink:href=http://docs.aws.amazon.com/cli/latest/reference/ecs/create-cluster.html">create</link>
+      the cluster. In the First Run Wizard select
+      <emphasis role="strong">Custom</emphasis> if you want to create a
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Is this a GUI menu item, an option, or a command? Try to use a more specific element (`<guimenu>`, `<option>`, etc.)

File: xml/art_sle_amazon_ecs.xml [hunk 3]
```diff
@@ -41,21 +41,27 @@
...
+      setup unique to your own needs. The
+      <emphasis role="strong">Amazon ECS sample</emphasis> selection creates
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Seems `<guimenu>` would fit here.

File: xml/art_sle_amazon_ecs.xml [hunk 4]
```diff
@@ -41,21 +41,27 @@
...
-      further configuration of the instance.
-    </para>
-    <para>
-      Renaming the cluster requires additional configuration once an ECS
-      instance is running. When the cluster name is
-      <emphasis role="strong">default</emphasis> the ECS instance will
-      automatically attach itself to the cluster. The additional configuration
-      steps for clusters with a name other than
-      <emphasis role="strong">default</emphasis> are described in the
+      <emphasis role="strong">default</emphasis>. When the cluster is named
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Better use `<systemitem>`.

File: xml/art_sle_amazon_ecs.xml [hunk 5]
```diff
@@ -41,21 +41,27 @@
...
+      <emphasis role="strong">default</emphasis> an ECS instance will
+      automatically attach itself to the cluster without additional
+      configuration steps. Selecting a custom name for the cluster
+      requires the instance to be configured as outlined in the
       <xref linkend="alternative-cluster-name"/> section below.
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: In some cases, the `<xref/>` creates already the word "section". As such, I would rephrase the sentence to avoid that issue:

```
Selecting a custom name for the cluster requires the instance to be configured
as outlined in <xref linkend="alternative-cluster-name"/>.
```

File: xml/art_sle_amazon_ecs.xml [hunk 6]
```diff
@@ -132,6 +138,69 @@
           cluster. Remember to select the ECS role when launching the
           instance.
         </para>
+        <para>
+          You may use the Amazon EC2 Web Console to launch instances.
+          The SUSE Linux Enterprise Server ECS images are found in the
+          so called General Catalog. From the
+          <emphasis role="strong">Instances</emphasis> view select
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Seems `<guimenu>` should fit here.

File: xml/art_sle_amazon_ecs.xml [hunk 7]
```diff
@@ -132,6 +138,69 @@
...
+          <emphasis role="strong">Launche Instance</emphasis>. In the left
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Same here.

File: xml/art_sle_amazon_ecs.xml [hunk 8]
```diff
@@ -132,6 +138,69 @@
...
+          column select the <emphasis role="strong">Community AMIs</emphasis>
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Again.

File: xml/art_sle_amazon_ecs.xml [hunk 9]
```diff
@@ -132,6 +138,69 @@
...
+          tab. Check the <emphasis role="strong">SUSE Linux</emphasis> box
+          on the left and enter <emphasis role="strong">ecs</emphasis> into
+          the search box. When configuring the instance parameters in the
+          <emphasis role="strong">Configure Instance Details</emphasis> view
+          select the <emphasis role="strong">ecsInstanceRole</emphasis> as
+          the IAM role. Then launch the instance.
+        </para>
+        <para>
+          Alternatively you can use the aws command line tool to
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Wrap the word "aws" with the `<command>` element.

File: xml/art_sle_amazon_ecs.xml [hunk 10]
```diff
@@ -132,6 +138,69 @@
...
+          <link xlink:href="http://docs.aws.amazon.com/cli/latest/reference/ec2/run-instances.html">run</link>
+          instances.
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Colon after this sentence instead of a full stop.

File: xml/art_sle_amazon_ecs.xml [hunk 11]
```diff
@@ -132,6 +138,69 @@
...
+        </para>
+        <screen>$ aws ec2 run-instances --image-id IMAGE_ID_SEE_PINT_OUTPUT --region EC2_REGION_NAME --key-name SSH_KEY_PAIR_NAME --instance-type DESIRED_INSTANCE_TYPE --subnet-id SUBNET_ID --iam-instance-profile 'Arn=PROFILE_PATH'</screen>
+        <para>
+          The PROFILE_PATH may be obtained by querying the IAM service.
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Is PROFILE_PATH a placeholder? A variable name?

* for placeholders, use `<replaceable>`
* for a variable name, use `<varname>`

File: xml/art_sle_amazon_ecs.xml [hunk 12]
```diff
@@ -132,6 +138,69 @@
...
+        <para>
+        <screen>$ aws iam list-instance-profiles</screen>
+        <para>
+          The SUBNET_ID will depend on your network configuration. If you
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Same here: SUBNET_ID placeholder, variable name, environment variable?

File: xml/art_sle_amazon_ecs.xml [hunk 13]
```diff
@@ -132,6 +138,69 @@
...
+          created your setup using the First Run Wizard you will most likely
+          want the instance to run in the subnet created by the wizard. Thus,
+          you can obtain the value by inspecting the properties of the running
+          ECS Instance.
+        </para>
+        <para>
+          Once the instance is running and if your cluster is named
+          <emphasis role="strong">default</emphasis> you can confirm that
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Use `<systemitem>` instead of `<emphasis>`.

File: xml/art_sle_amazon_ecs.xml [hunk 14]
```diff
@@ -184,5 +253,46 @@
...
+      container host instance will not be able to attach itself to your
+      cluster. Should you end up in a situation where the instance does not
+      attach itself to your <emphasis role="strong">default</emphasis>
+      cluster, you can modify your network configuration to ensure a route
+      to Amazon S3 and then simply log into your container instance as start
+      the service as follows:
+    </para>
+    <screen>$ systemctl start amazon-ecs.service</screen>
+ </sect1>
+ <sect1 xml:id="using sles containers">
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: No spaces in IDs please. :-)

File: xml/art_sle_amazon_ecs.xml [hunk 15]
```diff
@@ -21,9 +21,9 @@
...
    <para>
-    Amazon EC2 Container Service (ECS) is a container management service
-    that provides a set of APIs for scheduling container workloads
-    across EC2 clusters.
+    <link xlink:href="https://aws.amazon.com/ecs/">Amazon EC2 Container Service (ECS)</link>
+    is a container management service that provides a set of APIs for
+    scheduling container workloads across EC2 clusters.
    </para>
    <para>
     SUSE Linux Enterprise 12 images with the
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Replace "SUSE Linux Enterprise" with the entity `&sle;`.

You can find additional entity definitions in the file `entity-decl.ent`.

File: xml/art_sle_amazon_ecs.xml [hunk 16]
```diff
@@ -132,6 +138,69 @@
           cluster. Remember to select the ECS role when launching the
           instance.
         </para>
+        <para>
+          You may use the Amazon EC2 Web Console to launch instances.
+          The SUSE Linux Enterprise Server ECS images are found in the
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Replace "SUSE Linux Enterprise Server" with the entity `&sls;`.

File: xml/art_sle_amazon_ecs.xml [hunk 17]
```diff
@@ -132,6 +138,69 @@
...
+          ECS Instance.
+        </para>
+        <para>
+          Once the instance is running and if your cluster is named
+          <emphasis role="strong">default</emphasis> you can confirm that
+          the instance has attached itself to the cluster as follows:
+        </para>
+        <screen>$ aws ecs describe-clusters</screen>
+        <para>
+          The <emphasis role="strong">registeredContainerInstancesCount</emphasis>
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Probably `<guimenu>` instead of `<emphasis>`.

File: xml/art_sle_amazon_ecs.xml [hunk 18]
```diff
@@ -184,5 +253,46 @@
...
+      Once configured installing the docker image is simple:
+    </para>
+    <screen>$ zypper se docker-image</system>
+    <para>
+      Install the image you are interested in, the SUSE Linux Enterprise
+      Server 12 image for example:
+    </para>
+    <screen>$ zypper in sles12-docker-image</screen>
+    <para>
+      The SUSE Linux Enterprise Server EULA does not allow you to push this
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Not sure if we have an "EULA". Better name it just "license"? EULA sounds a bit Microsoft-ish. :grinning:

File: xml/art_sle_amazon_ecs.xml (hunk 1 above)
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: "... the aws command line tool**s**" ?

Is it really plural? From my perspective and as far as I can see, there is only one `aws` command (singular!) but with several subcommands (plural!).
Probably a matter of definition, just stumbled upon.

File: xml/art_sle_amazon_ecs.xml (hunk 17 above)
Comment (rjschwei) [inline xml/art_sle_amazon_ecs.xml]: The "registeredContainerInstancesCount" is an entry in the json output produced by the command given in the <screen> section just prior to this. I think emphasis as in "look for this" are appropriate.

File: xml/art_sle_amazon_ecs.xml [hunk 19]
```diff
@@ -132,6 +138,69 @@
...
+        </para>
+        <para>
+          Once the instance is running and if your cluster is named
+          <emphasis role="strong">default</emphasis> you can confirm that
+          the instance has attached itself to the cluster as follows:
+        </para>
+        <screen>$ aws ecs describe-clusters</screen>
+        <para>
+          The <emphasis role="strong">registeredContainerInstancesCount</emphasis>
+          entry in the output provides the number of instances considered to
```
Comment (sknorr) [inline xml/art_sle_amazon_ecs.xml]: ... _is_ considered to ...

File: xml/art_sle_amazon_ecs.xml (hunk 17 above)
Comment (sknorr) [inline xml/art_sle_amazon_ecs.xml]: `<literal/>` might be the best idea, then...

File: xml/art_sle_amazon_ecs.xml [hunk 20]
```diff
@@ -21,17 +21,17 @@
...
-    Amazon EC2 Container Service (ECS) is a container management service
-    that provides a set of APIs for scheduling container workloads
-    across EC2 clusters.
+    <link xlink:href="https://aws.amazon.com/ecs/">Amazon EC2 Container Service (ECS)</link>
+    is a container management service that provides a set of APIs for
+    scheduling container workloads across EC2 clusters.
    </para>
    <para>
-    SUSE Linux Enterprise 12 images with the
+    <productname>&sle</productname> 12 images with the
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sle;` (semicolon missing)

File: xml/art_sle_amazon_ecs.xml [hunk 21]
```diff
@@ -21,17 +21,17 @@
...
     <emphasis role="strong">ecs</emphasis> TLA in the image name are designed
     to integrate with the Amazon EC2 Container service. The image contains
     an agent that interacts with Docker to start/stop new containers and
     gather information about running containers. The agent is fully
-    integrated with SUSE Linux Enterprise.
+    integrated with <productname>&sle</productname>.
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sle;` (semicolon missing)

File: xml/art_sle_amazon_ecs.xml [hunk 22]
```diff
@@ -41,22 +41,29 @@
     <para>
       Creating a new ECS cluster in Amazon is accomplished by using the
       <link xlink:href="https://console.aws.amazon.com/ecs/home#/firstRun">First
-      Run Wizard</link>
+      Run Wizard.</link> Alternatively you may use the
+      <command>aws</command> command line tool to create the cluster, refer to 
+      <link xlink:href=http://docs.aws.amazon.com/cli/latest/reference/ecs/create-cluster.html">create</link> for details.
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Missing quote (") after `xlink:href`

File: xml/art_sle_amazon_ecs.xml [hunk 23]
```diff
@@ -41,22 +41,29 @@
...
+      In the First Run Wizard select
+      <guibutton>Custom</guibutton> if you want to create a
+      setup unique to your own needs. The
+      <guibutton>Amazon ECS sample</guibutton> selection creates
+      a pre-defined web application setup, that may be customized. By default
+      the First Run Wizard selects the Amazon ECS AMI, we will change this
+      later. 
+      <note>Note that using the First Run Wizard also creates an autoscaling
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: This isn't a valid DocBook construct. I would recommend the following:

1. Close the paragraph after "later.":

    ```
... we will change this later.</para>
```

1. Open a new note:

   ```
<note>
    <!-- <title>...</title> --><!-- Optional title -->
    <para>Using the First Run  Wizard also creates ... </para>
</note>
```

1. Don't forget to remove the closing `</para>`. ;-)

File: xml/art_sle_amazon_ecs.xml [hunk 24]
```diff
@@ -105,13 +112,13 @@
       ECS specific policies and roles can be found in the Amazon documentation:
       <link xlink:href="http://docs.aws.amazon.com/AmazonECS/latest/developerguide/instance_IAM_role.html">IAM Role</link>
     </para>
-    <para>
+    <note>
       Note that the policy given above is more permissive than the
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Wrap the text inside `<para>...</para>` as this isn't a valid DocBook construct.

File: xml/art_sle_amazon_ecs.xml [hunk 25]
```diff
@@ -105,13 +112,13 @@
...
       policy example provided in the Amazon documentation.
-    </para>
+    </note>
  </sect1>
  <sect1 xml:id="attach-sles12-ecs-instances-to-the-cluster">
-    <title>Attach SLES12 ECS instances to the cluster</title>
+    <title>Attach <productname>&sls</productname> ECS instances to the cluster</title>
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sls;` (add a semicolon)

File: xml/art_sle_amazon_ecs.xml [hunk 26]
```diff
@@ -120,7 +127,7 @@
     <orderedlist>
       <listitem>
         <para>
-          Find the latest SLES12 ECS ami using
+          Find the latest <productname>&sls</productname> ECS ami using
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sls;` (add a semicolon)

File: xml/art_sle_amazon_ecs.xml [hunk 27]
```diff
@@ -132,6 +139,69 @@
           cluster. Remember to select the ECS role when launching the
           instance.
         </para>
+        <para>
+          You may use the Amazon EC2 Web Console to launch instances.
+          The <productname>&sls</productname> ECS images are found in the
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sls;` (add a semicolon)

File: xml/art_sle_amazon_ecs.xml [hunk 28]
```diff
@@ -132,6 +139,69 @@
...
+          <emphasis role="strong">Configure Instance Details</emphasis> view
+          select the <guimenuitem>ecsInstanceRole</guimenuitem> as
+          the IAM role. Then launch the instance.
+        </para>
+        <para>
+          Alternatively you can use the <command>aws</command> command line
+          tool to run instances, see
+          <link xlink:href="http://docs.aws.amazon.com/cli/latest/reference/ec2/run-instances.html">run-instances</link> for details.
+        </para>
+        <screen>$ aws ec2 run-instances --image-id <replaceable>IMAGE_ID_SEE_PINT_OUTPUT</replaceable> --region <replaceable>EC2_REGION_NAME</replaceable> --key-name <replaceable>SSH_KEY_PAIR_NAME</replaceable> --instance-type <replaceable>DESIRED_INSTANCE_TYPE</replaceable> --subnet-id <replaceable>SUBNET_ID</replaceable> --iam-instance-profile 'Arn=<replaceable>PROFILE_PATH</replaceable>'</screen>
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Technically it's absolutely correct. However, IMHO it would be better to introduce some linebreaks to improve readability, for example:

```
<screen>$ aws ec2 run-instances --image-id <replaceable>IMAGE_ID_SEE_PINT_OUTPUT</replaceable> \
    --region <replaceable>EC2_REGION_NAME</replaceable> \
  [...]</screen>
```

File: xml/art_sle_amazon_ecs.xml [hunk 29]
```diff
@@ -132,6 +139,69 @@
...
+          <systemitem>default</systemitem> you can confirm that
+          the instance has attached itself to the cluster as follows:
+        </para>
+        <screen>$ aws ecs describe-clusters</screen>
+        <para>
+          The <literal>registeredContainerInstancesCount</literal>
+          entry in the output provides the number of instances considered to
+          be part of the cluster. If you used the First Run Wizard to create
+          the cluster the number should be at least 2 after you launch the
+          <productname>&sls</productname> ECS instance.
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sls;` (add a semicolon)

File: xml/art_sle_amazon_ecs.xml [hunk 30]
```diff
@@ -132,6 +139,69 @@
...
+        </para>
+      </listitem>
+      <listitem>
+        <para>
+          If you used the First Run Wizard you already have an
+          <link xlink:href="https://aws.amazon.com/autoscaling/>Auto Scaling</link>
+          group. You can
+          <link xlink:href="http://docs.aws.amazon.com/cli/latest/reference/autoscaling/attach-instances.html">add</link>
+          the running <productname>&sls</productname> instance to the existing
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sls;` (add a semicolon)

File: xml/art_sle_amazon_ecs.xml [hunk 31]
```diff
@@ -184,5 +254,83 @@
...
+      cluster. Should you end up in a situation where the instance does not
+      attach itself to your <emphasis role="strong">default</emphasis>
+      cluster, you can modify your network configuration to ensure a route
+      to Amazon S3 and then simply log into your container instance as start
+      the service as follows:
+    </para>
+    <screen>$ systemctl start amazon-ecs.service</screen>
+ </sect1>
+ <sect1 xml:id="using-sles-containers">
+    <title>Using <productname>&sls</productname> Containers</title>
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sls;` (add a semicolon)

File: xml/art_sle_amazon_ecs.xml [hunk 32]
```diff
@@ -184,5 +254,83 @@
...
+    <para>
+      <productname>&sls</productname> Docker images are not available on
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sls;` (add a semicolon)

File: xml/art_sle_amazon_ecs.xml [hunk 33]
```diff
@@ -184,5 +254,83 @@
...
+      Docker Hub. However, SUSE provides pre-build docker images as
+      packages in the SLE-Module-Containers12-Updates repository. Given
+      that your launched instance has a route to the Internet and that
+      the traffic originates with an IP address assigned by EC2 the
+      instance will automatically register with an update server and
+      the SLE-Module-Containers12-Updates repository will be configured.
+      Once configured installing a <productname>&sls</productname> docker
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sls;` (add a semicolon)

File: xml/art_sle_amazon_ecs.xml [hunk 34]
```diff
@@ -184,5 +254,83 @@
...
+      image is simple:
+    </para>
+    <screen>$ zypper se docker-image</system>
+    <para>
+      Install the image you are interested in, the
+      <productname>&sls</productname> 12 image for example:
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sls;` (add a semicolon)

File: xml/art_sle_amazon_ecs.xml [hunk 35]
```diff
@@ -184,5 +254,83 @@
...
+    </para>
+    <screen>$ zypper in sles12-docker-image</screen>
+    <para>
+      The <productname>&sls</productname> license does not allow you to
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sls;` (add a semicolon)

File: xml/art_sle_amazon_ecs.xml [hunk 36]
```diff
@@ -184,5 +254,83 @@
...
+      push this image to a public docker registry. However you can use the
+      Amazon hosted private
+      <link xlink:href="https://aws.amazon.com/ecr/">Container Registry</link>
+      to store <productname>&sls</productname> containers.
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sls;` (add a semicolon)

File: xml/art_sle_amazon_ecs.xml [hunk 37]
```diff
@@ -184,5 +254,83 @@
...
+    </para>
+    <para>
+      For additional details about <productname>&sls</productname> Docker
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sls;` (add a semicolon)

File: xml/art_sle_amazon_ecs.xml [hunk 38]
```diff
@@ -184,5 +254,83 @@
...
+      images see section 6 of the
+      <link xlink:href="https://www.suse.com/documentation/sles-12/dockerquick/data/dockerquick.html#sec_docker_sle_images">Quick Start Guide</link>.
+    </para>
+ </sect1>
+ <sect1 xml:id="image-storage">
+   <title>Docker Image Storage</title>
+   <para>
+     The <productname>&sls</productname> ECS image uses btrfs as the storage
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: It's `&sls;` (add a semicolon)

File: xml/art_sle_amazon_ecs.xml [hunk 39]
```diff
@@ -184,5 +254,83 @@
...
+     root volume to store the docker images.
+   </para>
+   <para>
+     If you store a large number of docker images or you want to safeguard
+     your local docker images from accidental deletion you may consider
+     to place the image storage onto an external
+     <link xlink:href="https://aws.amazon.com/ebs/">Elastic Block Store</link>
+     device. If you format the device with btrfs no changes to the docker
+     configuration are required.
+     <note>
```
Comment (tomschr) [inline xml/art_sle_amazon_ecs.xml]: Same as in my previous comment (`<note>` should be stay outside of `<para>`).
